kong/types.go

```diff
@@ -146,6 +146,47 @@ type Healthcheck struct {
 	Threshold *float64            `json:"threshold,omitempty" yaml:"threshold,omitempty"`
 }
 
+// HealthDataAddress represents the health data address of a target
+// +k8s:deepcopy-gen=true
+type HealthDataAddress struct {
+	Port   *int    `json:"port,omitempty" yaml:"port,omitempty"`
+	IP     *string `json:"ip,omitempty" yaml:"ip,omitempty"`
+	Health *string `json:"health,omitempty" yaml:"data,omitempty"`
+	Weight *int    `json:"weight,omitempty" yaml:"weight,omitempty"`
+}
+
+// HealthDataWeight represents the health data weight of a target
+// +k8s:deepcopy-gen=true
+type HealthDataWeight struct {
+	Total       *int `json:"total,omitempty" yaml:"total,omitempty"`
+	Available   *int `json:"available,omitempty" yaml:"available,omitempty"`
+	Unavailable *int `json:"unavailable,omitempty" yaml:"unavailable,omitempty"`
+}
+
+// HealthData represents the health data of a target
+// +k8s:deepcopy-gen=true
+type HealthData struct {
+	Host       *string             `json:"host,omitempty" yaml:"host,omitempty"`
+	Port       *int                `json:"port,omitempty" yaml:"port,omitempty"`
+	NodeWeight *int                `json:"nodeWeight,omitempty" yaml:"nodeWeight,omitempty"`
+	Weight     *HealthDataWeight   `json:"weight,omitempty" yaml:"weight,omitempty"`
+	Addresses  []HealthDataAddress `json:"addresses,omitempty" yaml:"addresses,omitempty"`
+	DNS        *string             `json:"dns,omitempty" yaml:"dns,omitempty"`
+}
+
+// UpstreamNodeHealth represents the node health of a upstream
+// +k8s:deepcopy-gen=true
+type UpstreamNodeHealth struct {
+	ID        *string     `json:"id,omitempty" yaml:"id,omitempty"`
+	CreatedAt *float64    `json:"created_at,omitempty" yaml:"created_at,omitempty"`
```

Comment:
time stamps are generally typed as `int` within this project. can we use that here as well?

Reply:
I was using at the beginning an `int`, however it is a `float64` that is returned.

By example:
```json
{
    "data": {
        "host":"10.0.0.1",
        "port":80,
        "weight":{
            "total":100,
            "available":100,
            "unavailable":0
        },
        "nodeWeight":100,
        "addresses":[
            {
                "port":80,
                "ip":"10.0.0.1",
                "health":"HEALTHCHECKS_OFF",
                "weight":100
            }
        ],
        "dns":"A"
    },
    "created_at":1608643189.157,
    "health":"HEALTHCHECKS_OFF",
    "id":"8f78d3dd-fcf3-4d9d-95d3-bce057ec13f5",
    "tags":null,
    "weight":100,
    "target":"10.0.0.1:80",
    "upstream":{
        "id":"f97522e0-2989-4dd2-8408-1eb390b65369"
    }
}
```

In this example, `created_at` has the value `1608643189.157` which is a float value.

Also it is inline with `Target` structure: https://github.com/Kong/go-kong/blob/main/kong/types.go#L172

Reply:
Make sense! thanks :)